pkg/clicmd/prototype_use.go

```diff
@@ -84,34 +87,80 @@ ks prototype use single-port-deployment nginx-depl \
 `
 )
 
-func newGenerateCmd(a app.App) *cobra.Command {
-	cmd := newPrototypeUseCmd(a)
+func newGenerateCmd(fs afero.Fs) *cobra.Command {
+	cmd := newPrototypeUseCmd(fs)
 	cmd.Use = "generate <prototype-name> <component-name> [type] [parameter-flags]"
 
 	return cmd
 }
 
-func newPrototypeUseCmd(a app.App) *cobra.Command {
+func newPrototypeUseCmd(fs afero.Fs) *cobra.Command {
 	prototypeUseCmd := &cobra.Command{
-		Use:                "use <prototype-name> <componentName> [type] [parameter-flags]",
-		Short:              protoShortDesc["use"],
-		Long:               prototypeUseLong,
-		Example:            prototypeUseExample,
+		Use:     "use <prototype-name> <componentName> [type] [parameter-flags]",
+		Short:   protoShortDesc["use"],
+		Long:    prototypeUseLong,
+		Example: prototypeUseExample,
+		FParseErrWhitelist: cobra.FParseErrWhitelist{
+			UnknownFlags: true,
+		},
 		DisableFlagParsing: true,
-		RunE: func(cmd *cobra.Command, rawArgs []string) error {
-			if len(rawArgs) == 1 && (rawArgs[0] == "--help" || rawArgs[0] == "-h") {
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.DisableFlagParsing = false
+			err := cmd.ParseFlags(args)
+
+			if err != nil {
+				return cmd.FlagErrorFunc()(cmd, err)
+			}
+
+			if cmd.Flags().Lookup("help").Changed {
 				return cmd.Help()
 			}
 
+			args = removePersistentFlags(cmd.Flags(), args)
+
 			m := map[string]interface{}{
-				actions.OptionApp:       a,
-				actions.OptionArguments: rawArgs,
-				// We don't pass flagTLSSkipVerify because flag parsing is disabled
+				actions.OptionFs:        fs,
+				actions.OptionArguments: args,
 			}
+			addGlobalOptions(m)
 
 			return runAction(actionPrototypeUse, m)
 		},
 	}
 
 	return prototypeUseCmd
 }
+
+func removePersistentFlags(flags *pflag.FlagSet, args []string) []string {
+	flags.Visit(func(flag *pflag.Flag) {
+		for i, arg := range args {
+			if len(arg) == 0 || arg[0] != '-' || len(arg) == 1 {
+				continue
+			}
+
+			if arg[1] == '-' {
+				if len(arg) == 2 {
+					// "--" terminates the flags
+					break
+				}
+
+				// long flag that might be known by ks
+				splitFlag := strings.SplitN(arg[2:], "=", 2)
+				if splitFlag[0] == flag.Name {
+					if len(splitFlag) == 2 {
+						// global flag with =
+						args = append(args[:i], args[i+1:]...)
+					} else if flag.NoOptDefVal != "" {
+						// optional flag that sets default
+						args = append(args[:i], args[i+1:]...)
+					} else {
+						// '--flag arg' means remove two
+						args = append(args[:i], args[i+2:]...)
```

Comment:
Can this go out of bounds if args ends with `--flag` and no corresponding value?

Reply:
no - confirmed with `--tls-skip-verify` at end of command. but flow control wise we don't hit this else block unless we've been given a flag that requires an argument with its argument. because this runs after the original Parse we already know we have a valid command line apart from unknown flags (which we are ignoring anyway).